helper/file/file.go

```diff
@@ -0,0 +1,133 @@
+package file
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/samber/lo"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	obskit "github.com/rudderlabs/rudder-observability-kit/go/labels"
+	ht "github.com/rudderlabs/rudder-server/helper/types"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+var jsonfast = jsoniter.ConfigCompatibleWithStandardLibrary
+
+type fields struct {
+	Input    any         `json:"request"`
+	Output   any         `json:"response"`
+	Metainfo ht.MetaInfo `json:"metadata"`
+}
+
+type fileConfig struct {
+	MaxBatchSize    int
+	MaxBatchTimeout time.Duration
+}
+
+type FileHandle struct {
+	dir                 string
+	fileConfig          *fileConfig
+	logger              logger.Logger
+	fieldsChan          chan fields
+	backgroundCtxCancel context.CancelFunc
+	backgroundWait      func() error
+}
+
+func getFileConfig(parentMod string) *fileConfig {
+	maxBatchSzConfPath := fmt.Sprintf("%s.DebugHelper.maxBatchSize", parentMod)
+	maxBatchTimeoutConfPath := fmt.Sprintf("%s.DebugHelper.maxBatchTimeout", parentMod)
+
+	return &fileConfig{
+		MaxBatchTimeout: config.GetReloadableDurationVar(20, time.Second, maxBatchTimeoutConfPath).Load(),
+		MaxBatchSize:    config.GetReloadableIntVar(200, 1, maxBatchSzConfPath).Load(),
+	}
+}
+
+func WithDirectory(dir string) func(*FileHandle) {
+	return func(fh *FileHandle) {
+		fh.dir = dir
+	}
+}
+
+func WithConfig(maxBatchSz int, maxBatchTimeout time.Duration) func(*FileHandle) {
+	return func(fh *FileHandle) {
+		fh.fileConfig = &fileConfig{
+			MaxBatchSize:    maxBatchSz,
+			MaxBatchTimeout: maxBatchTimeout,
+		}
+	}
+}
+
+func New(parentMod string, opts ...func(*FileHandle)) (*FileHandle, error) {
+	fileConfig := getFileConfig(parentMod)
+	fieldsChan := make(chan fields, fileConfig.MaxBatchSize)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	g, _ := errgroup.WithContext(ctx)
+
+	handle := &FileHandle{
+		dir:                 "mydebuglogs/",
+		fileConfig:          fileConfig,
+		logger:              logger.NewLogger().Child("helper.file"),
+		fieldsChan:          fieldsChan,
+		backgroundCtxCancel: cancel,
+		backgroundWait:      g.Wait,
+	}
```

Comment:
We don't really have a use for context. Closing the channel is enough to stop the process.
```suggestion
	g, _ := errgroup.WithContext(context.Background())

	handle := &FileHandle{
		dir:                 "mydebuglogs/",
		fileConfig:          fileConfig,
		logger:              logger.NewLogger().Child("helper.file"),
		fieldsChan:          fieldsChan,
		backgroundWait:      g.Wait,
	}
```